module6/src/main/java/Main.java

```diff
@@ -1,37 +1,73 @@
 import bean.CalculationService;
 import bean.DirectoriesScanner;
-import bean.FileReader;
-import writer.WriterWrapper;
+import bean.ReportReader;
+import writer.ReportWriter;
 
 import java.io.File;
 import java.io.IOException;
 import java.util.List;
 
 public class Main {
-
+    /**
+     * Create .txt file with direction tree from direction path, or read information from file
+     * and calculate number files and directories {@link CalculationService} in this file
+     *
+     * @param args can contains path to .txt file {@link ReportReader#readFromFile(String)}
+     *             or to directory {@link Main#print(ReportWriter, String)} (String)}
+     */
     public static void main(String[] args) {
-        WriterWrapper writer = new WriterWrapper();
-        try {
-            writer.openWriter();
-            DirectoriesScanner.createReportInFile(0, new File("d:/Rammstein"), writer);
-        } catch (IOException e) {
-            e.printStackTrace();
+        List<String> stringList;
+        String path = args[0];
+        if (isArgumentAFile(path)) {
+            try {
+                stringList = ReportReader.readFromFile(path);
+                Calculate(stringList);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            readFilesAndWriteDirectoryTree(path);
         }
-        List<String> stringList = null;
-        try {
-            stringList = FileReader.readFromFile();
-        } catch (IOException e) {
-            e.printStackTrace();
+    }
+
+    private static boolean isArgumentAFile(String path) {
+        boolean bool = false;
+        if (path.endsWith(".txt")) {
+            bool = true;
         }
+
+        return bool;
+    }
+
+    private static void readFilesAndWriteDirectoryTree(String path) {
+        ReportWriter writer = new ReportWriter();
+        print(writer, path);
```

Comment:
You can create ReportWriter object in the method `print`.
Also you can get rid from method `print` because its existence doesn't bring additional sence.